diff -uraN ../libaal-1.0.5/include/aal/types.h ./include/aal/types.h
--- ../libaal-1.0.5/include/aal/types.h	2005-07-28 21:13:41.000000000 +0200
+++ ./include/aal/types.h	2020-01-28 22:26:25.709801529 +0100
@@ -9,6 +9,8 @@
 #ifdef HAVE_CONFIG_H
 #  include <config.h>
 #endif
+#include <sys/types.h>
+#include <stdint.h>
 
 /* Some usefull return codes */
 #undef EINVAL
@@ -31,17 +33,17 @@
 typedef signed char             int8_t;
 typedef short int               int16_t;
 typedef int                     int32_t;
-__extension__
-typedef long long int           int64_t;
+/*__extension__
+typedef long long int           int64_t;*/
 #endif
  
 typedef unsigned char           uint8_t;
 typedef unsigned short int      uint16_t;
 #ifndef __uint32_t_defined
 #define __uint32_t_defined
-typedef unsigned int            uint32_t;
+/*typedef unsigned int            uint32_t;
 __extension__
-typedef unsigned long long int  uint64_t;
+typedef unsigned long long int  uint64_t;*/
 #endif
 
 #define MAX_UINT8  ((uint8_t)~0)
